node-tree-sitter.d.ts

```diff
@@ -0,0 +1,92 @@
+declare module "tree-sitter" {
+  export class Document {
+    // Getters
+    public rootNode: AstNode | null;
+
+    // Methods
+    public setLanguage(module: any): void;
+    public setInputString(input: string): void;
+    public parse(): void;
+    public getLogger(): LoggerFunc;
+    public setLogger(logFunc: LoggerFunc): void;
+    public getInput(): StringInput | null;
+    public setInput(input: StringInput): void;
+    public edit(delta: EditDelta): Document;
+    public invalidate(): void;
+  }
+
+  // TODO Need to specify the arguments
+  type LoggerFunc = (...args: any[]) => void;
```

Comment:
The logger function takes 3 arguments: a message string, an object containing arbitrary key/value parameters (where the values are strings), and a string that's either `parse` or `lex`, depending on which component emitted the message.

So maybe this:

```ts
type LoggerFunc = (message: string, params: {[param: string]: string}, type: "parse" | "lex") => void;
```

Reply:
Looks good! I'll update the PR, thanks =]